stdlib/src/builtin/object.mojo

```diff
@@ -75,6 +76,22 @@ struct _ImmutableString:
             return 0
         return -1 if self.length < rhs.length else 1
 
+struct _List(Movable):
```

Comment:
`List[_ObjectImpl]()` itself is not moveable, so I create another movable struct

Reply:
**Question** Is this needed?  `_ObjectImpl` should implicitly conform to `Movable` trait since it is `register_passable`.  That's why the pre-existing code works fine with `List[_ObjectImpl]` today, right?

I think for this PR, it should work if we can just augment `_RefCountedList` struct to use an `Arc` internally and remove the internal `refcount` data member.  Does that not work for some reason?

Reply:
I retry it and it works. I previously saw the error saying that `List[_ObjectImpl]` is not movable, which is why I wrapped it into another movable struct. Maybe it is fixed in the new version 🤔